app/models/lookup_keys/lookup_key.rb

```diff
@@ -21,14 +21,13 @@ class LookupKey < ActiveRecord::Base
 
   alias_attribute :value, :default_value
   before_validation :cast_default_value
-  after_validation :reset_override_params, :if => ->(key) { key.override_changed? && !key.override? }
 
   validates :key, :presence => true
   validates :validator_type, :inclusion => { :in => VALIDATOR_TYPES, :message => N_("invalid")}, :allow_blank => true, :allow_nil => true
   validates :key_type, :inclusion => {:in => KEY_TYPES, :message => N_("invalid")}, :allow_blank => true, :allow_nil => true
   validate :validate_default_value
   validates_associated :lookup_values
-  validate :disable_merge_overrides, :disable_avoid_duplicates, :disable_merge_default, :if => :override?
+  validate :disable_merge_overrides, :disable_avoid_duplicates, :disable_merge_default, :if =>  Proc.new { |lk| lk.override || !lk.puppet? }
```

Comment:
Doesn't have to be in this pr, but I would consider combining these to one custom validator and including it with a different condition in the children, which would get rid of complex condition and separate concerns better.